### PR TITLE
Fork choice update implemented

### PR DESCRIPTION
**Pull request by enriavil1**

### CODE REVIEW

File: silkrpc/ethbackend/backend_grpc.cpp [hunk 1]
```diff
@@ -93,21 +93,29 @@ asio::awaitable<PayloadStatus> BackEndGrpc::engine_new_payload_v1(ExecutionPaylo
...
-    }
+    PayloadStatus payload_status = decode_payload_status(reply);
     SILKRPC_DEBUG << "BackEnd::engine_new_payload_v1 data=" << payload_status << " t=" << clock_time::since(start_time) << "\n";
     co_return payload_status;
 }
 
+asio::awaitable<ForkchoiceUpdatedReply> BackEndGrpc::engine_forkchoice_updated_v1(ForkchoiceUpdatedRequest forkchoice_updated_request) {
+    const auto start_time = clock_time::now();
+    EngineForkChoiceUpdatedV1Awaitable fcu_awaitable{executor_, stub_, queue_};
+    auto req{encode_forkchoice_updated_request(forkchoice_updated_request)};
```
Comment (canepat) [inline silkrpc/ethbackend/backend_grpc.cpp:104]: Maybe `req` can be declared `const` here?

**Review by canepat — CHANGES_REQUESTED** [review]
Please add also an integration test for `engine_forkchoiceUpdatedV1` API

File: silkrpc/commands/engine_api.cpp [hunk 2]
```diff
@@ -74,6 +74,46 @@ asio::awaitable<void> EngineRpcApi::handle_engine_new_payload_v1(const nlohmann:
     #endif
 }
 
+// Format for params is a JSON list containing two objects
+// one ForkchoiceState and one PayloadAttributes, i.e. [ForkchoiceState, PayloadAttributes]
```
[the remark this answers is not in the document]
Comment (enriavil1) [reply silkrpc/commands/engine_api.cpp:78]: Will do!

File: silkrpc/commands/engine_api.cpp [hunk 3]
```diff
@@ -74,6 +74,46 @@ asio::awaitable<void> EngineRpcApi::handle_engine_new_payload_v1(const nlohmann:
...
+            const PayloadAttributes payload_attributes = params[1].get<PayloadAttributes>();
+            const ForkchoiceUpdatedRequest forkchoice_update_request{
+                .forkchoice_state = forkchoice_state,
+                .payload_attributes = std::optional<PayloadAttributes>{payload_attributes}
+            };
+            reply = co_await backend_->engine_forkchoice_updated_v1(forkchoice_update_request);
+        } else {
+            const ForkchoiceUpdatedRequest forkchoice_update_request{
+                .forkchoice_state = forkchoice_state,
+                .payload_attributes = std::nullopt
```
[the remark this answers is not in the document]
Comment (enriavil1) [reply silkrpc/commands/engine_api.cpp:102]: Thank you!

File: silkrpc/ethbackend/backend_grpc.cpp (hunk 1 above)
Comment (enriavil1) [reply silkrpc/ethbackend/backend_grpc.cpp:104]: yeah

**Review by canepat — APPROVED** [review]

**Pull request merged**